### DISCUSSION TITLE
Pass link as variable to slack message payload

### DISCUSSION THREAD

**Discussion by you1anna** [fastlane/fastlane#17849] [Q&A] [answered]
Is there a way to pass a link variable to a slack post? The url needs to be constructed first, which cannot be done within 
`attachment_properties` json, so it currently renders as plain text.

https://docs.fastlane.tools/actions/slack/

Fastfile:

downloadlink = GOCD_URL + '/go/files/' + ENV["GO_PIPELINE_NAME"] + '/xyz'

```ruby 
slack(
  slack_url: "https://hooks.slack.com/services/myhook",
  message: "Staging app successfully built!",
  channel: "#r-alerts-test",
  success: true,
  payload: { 
    "Environment" => "Staging",
  },  
  default_payloads: [:git_branch, :git_author],
  attachment_properties: {
    thumb_url: "https://play-store/xxx",
    fields: [
      {
      title: ".apk file",
      value: "<downloadlink|Download>",
      short: false
    }
  ]
  }
)
```

The output post for `downloadlink` is plain text 

I was referring to https://api.slack.com/messaging/composing/layouts#attachments 

Has anyone found a way to make `downloadlink` variable work as a link in the Slack post?

**Comment by you1anna** [comment, marked as answer]
I found the answer, variables need to use #{link} syntax in this case.

**Reply by iamjamal** [reply]
Hey @you1anna ... Can you paste your solution please. 
Your answer is the only solution I could find.
I need to send a AppCenter build link to Slack via Fastlane. Need help!
Thanks!

**Comment by marcosgqb** [comment]
if there is still the error or if someone needs it, try in this format:
'"${downloadlink}"'
worked for me

**Reply by alekseytivonchik** [reply]
My solution in Fastfile
```
attachment_properties: {
        fields: [
          {
            title: "Pipeline",
            value: "<https://example.com/-/pipelines/" + ENV['CI_PIPELINE_ID'] + "|" + ENV['CI_PIPELINE_ID'] + ">",
            short: true
          }
        ]
}
```
